### PR TITLE
Implement arithmetic opcodes on the interpreter side

### PR DESCRIPTION
**Pull request by Nashtare**
This PR implements the remaining (signed) arithmetic operations on the interpreter's side.

### CODE REVIEW

File: evm/src/cpu/kernel/interpreter.rs [hunk 1]
```diff
@@ -809,6 +918,70 @@ impl<'a> Interpreter<'a> {
...
+
+    match (x_is_pos, y_is_pos) {
+        (true, true) => x.cmp(&y),
+        (true, false) => Ordering::Greater,
+        (false, true) => Ordering::Less,
+        (false, false) => x.cmp(&y).reverse(),
+    }
+}
+
+const MINUS_ZERO: U256 = U256([
```
Comment (dlubarov) [inline evm/src/cpu/kernel/interpreter.rs:957]: Hm the yellowpaper mentions two's complement. If I'm not mistaken, only the all-zero-bits encoding is interpreted as zero, while this value would be interpreted as 2^-255 (which is special in that there's no positive 2^255).

Comment (Nashtare) [reply evm/src/cpu/kernel/interpreter.rs:957]: Ah mb. I'll adapt the methods taking into account the two's complements.

Comment (Nashtare) [reply evm/src/cpu/kernel/interpreter.rs]: I should have adapted all methods with two's complement!

Comment (Nashtare) [reply evm/src/cpu/kernel/interpreter.rs]: Should be good for review!

Comment (Nashtare) [reply evm/src/cpu/kernel/interpreter.rs]: I'll also add some tests for all those signed arith. operations in the related `tests` module, for consistency.

**Review by dlubarov — APPROVED** [review]
LGTM otherwise 👍

File: evm/src/cpu/kernel/interpreter.rs [hunk 2]
```diff
@@ -513,6 +577,43 @@ impl<'a> Interpreter<'a> {
...
+            let n = n.low_u64() as usize;
+            let num_bytes_prepend = 31 - n;
+
+            let mut x_bytes = [0u8; 32];
+            x.to_big_endian(&mut x_bytes);
+            let mut x_bytes = x_bytes[num_bytes_prepend..].to_vec();
+            let sign_bit = x_bytes[0] >> 7;
+
+            self.push(if sign_bit == 0 {
+                x_bytes.extend_from_slice(&vec![0; num_bytes_prepend]);
```
Comment (dlubarov) [inline evm/src/cpu/kernel/interpreter.rs:607]: It looks like here the sign bits are appended after x, should it be prepended before?

Comment (Nashtare) [reply evm/src/cpu/kernel/interpreter.rs]: Yes you're right, I've messed up the ordering here. Sorry about that, fixed!

**Pull request merged**